lib/ansible/modules/database/misc/kibana_saved_object.py

```diff
@@ -0,0 +1,446 @@
+        timeout=timeout
+    )
+    try:
+        return fetch_url(module, request_params['url'],
+                         method='PUT',
+                         headers=request_params['headers'],
+                         data=request_params['data'],
+                         timeout=timeout)
+    except Exception as e:
+        module.fail_json(msg="An error occured while trying to get the object '{0}'. {1}".format(object_id, to_native(e)))
+
+
+def delete_object(module, object_id, object_type, kibana_url,
+                  timeout, tenant=None):
+
+    request_params = get_request_params(
+        module=module,
+        object_id=object_id,
+        object_type=object_type,
+        kibana_url=kibana_url,
+        tenant=tenant,
+        timeout=timeout,
+    )
+    try:
+        return fetch_url(module, request_params['url'],
+                         method='DELETE',
+                         headers=request_params['headers'],
+                         timeout=timeout)
+    except Exception as e:
+        module.fail_json(msg="An error occured while trying to get the object '{0}'. {1}".format(object_id, to_native(e)))
+
+
+def is_object_present(module, object_id, object_type, kibana_url,
+                      timeout, tenant=None):
+
+    r, r_info = get_object(
+        module=module,
+        object_id=object_id,
+        object_type=object_type,
+        kibana_url=kibana_url,
+        tenant=tenant,
+        timeout=timeout
+    )
+
+    try:
+        if r_info['status'] == 404:
+            return False, None
+        elif r_info['status'] == 200:
+            return True, r.read()
+        else:
+            module.fail_json(msg="Got status other then 200 or 404. {0}".format(r_info))
+    except Exception as e:
+        module.fail_json(msg="An error occured while trying to get the object '{0}'. {1}".format(object_id, to_native(e)))
+
+
+def main():
+    argument_spec = url_argument_spec()
+    del argument_spec['http_agent'], argument_spec['force'], argument_spec['url']
+    argument_spec.update(dict(
+        id=dict(type='str'),
+        type=dict(type='str', required=True,
+                  choices=[
+                      'visualization',
+                      'dashboard',
+                      'search',
+                      'index-pattern',
+                      'config',
+                      'timelion-sheet'
+                  ]),
+        state=dict(type='str', default='present', choices=['present', 'absent']),
+        content=dict(type='dict', required=True),
+        kibana_url=dict(type='str', required=True),
+        timeout=dict(type='int', default=30),
+        overwrite=dict(type='bool', default=False),
+        url_username=dict(type='str', aliases=['user']),
+        url_password=dict(type='str', aliases=['password'], no_log=True),
+
+        searchguard_tenant=dict(type='str'),
+    ))
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
```

Comment:
Your module claims to support check mode, but I don't see any reference to check_mode being implemented.

That said, it would be very useful to actually implement check_mode.

Reply:
Done.